assets/js/modules/analytics/components/setup/SetupForm.js

```diff
@@ -79,6 +81,22 @@ export default function SetupForm( { finishSetup } ) {
 		[ finishSetup, setValues, submitChanges ]
 	);
 
+	const isTagManagerAvailable = useSelect( ( select ) =>
+		select( CORE_MODULES ).isModuleAvailable( 'tagmanager' )
+	);
+	// Preloading the `live-container-version` call is necessary to display
+	// the loading state until the data is fetched from the server.
+	// This call is being made in the child component ExistingGTMPropertyNotice
+	// to ensure that the loading state is displayed before the data is available.
+	useSelect(
+		( select ) =>
+			isTagManagerAvailable &&
+			select( MODULES_TAGMANAGER ).getSingleAnalyticsPropertyID()
+	);
+	const gtmContainersResolved = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS ).hasFinishedLoadingGTMContainers()
+	);
```

Comment:
I see that we repeat the same code in a couple of places. Let's probably refactor it a little bit to have less duplicate code and justify having these selectors. Let's do the following:

1. In every form we have identical `StoreErrorNotices` and `ExistingGTMPropertyNotice` elements that we can simply move to this `SetupForm` component and remove them from there:
   https://github.com/google/site-kit-wp/blob/75c82e34f0f19d9a59448aa69099312063424544/assets/js/modules/analytics/components/setup/SetupFormUA.js#L87-L92 https://github.com/google/site-kit-wp/blob/75c82e34f0f19d9a59448aa69099312063424544/assets/js/modules/analytics/components/setup/SetupFormGA4.js#L104-L108 https://github.com/google/site-kit-wp/blob/75c82e34f0f19d9a59448aa69099312063424544/assets/js/modules/analytics/components/setup/SetupFormGA4Transitional.js#L96-L100
1. Then update the `ExistingGTMPropertyNotice` component to receive the `gtmAnalyticsPropertyID` value via property instead of selecting it from the datastore.
1. Update `SetupForm` and `SettingsForm` to pass already selected single Analytics propertyID to the `ExistingGTMPropertyNotice` element via prop.

Reply:
That's a good suggestion, @eugene-manuilov 👍